### PR TITLE
Changing in module common-router 

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**
added factories, changed abstract router and abstract handler and other...

### CODE REVIEW

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 25.02.17.
+ */
+
+public class RouterFactory<T extends CommonRouterHandler>  {
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java]: @AlexKazmirchuk This class has the same functionality as HandlerFactory. Copy paste ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+		implements RequestHandler {
 
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		Request<T> request = new Request<>();
+		request.setHeader(msg.getHeader());
+
+		// todo check if working in future
+		Class<T> genericClass = (Class<T>) ((ParameterizedType) this.getClass()
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @AlexKazmirchuk make static final field.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
+				.getGenericSuperclass())
+				.getActualTypeArguments()[0];
+		JacksonDataMapper mapper = new JacksonDataMapper();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @AlexKazmirchuk Why new mapper? We already have module for this.

**Comment by Dominys** [comment]
@AlexKazmirchuk Fix issues.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java]: @Dominys I wanted to do abstract parent for both , and overriding getRouteKey method, but  I don`t know how to do it correctly (some problem with generics)

Comment (AlexKazmirchuk) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java]: @Dominys  I will try to fix it

Comment (Dominys) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java]: @AlexKazmirchuk Yes. Fix this. There should not be copy-paste.

Comment (AlexKazmirchuk) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java]: @Dominys fixed , I think

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractFactory.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 27.02.17.
+ */
+
+public abstract class AbstractFactory<T extends Handler> {
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractFactory.java]: @AlexKazmirchuk Where is interface for factory? Factory must be injected by interface.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java [hunk 5]
```diff
@@ -4,7 +4,7 @@
 import com.softgroup.common.protocol.Response;
 import org.springframework.beans.factory.annotation.Autowired;
 
-public abstract class AbstractRouterHandler<T extends Handler>
+public abstract class AbstractRouterHandler<T extends RequestHandler>
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java]: @AlexKazmirchuk why RequestHandler? Router can work with request handlers and with another routers.

**Comment by Dominys** [comment]
@AlexKazmirchuk make one more PR with tests for this changes.

**Pull request merged**